common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -368,4 +381,70 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecFileStoreDetector {
+        static final Method TO_PATH_METHOD;
+        static final Method GET_FILE_STORE_METHOD;
+        static final Method MOUNT_ENTRY_METHOD;
+        static final Method HAS_OPTION_METHOD;
+
+        static {
+            Method toPathMethod = null;
+            Method getFileStoreMethod = null;
+            Method mountEntryMethod = null;
+            Method hasOptionMethod = null;
+            try {
+                // Use reflection so Java 6 fails gracefully.
+                toPathMethod = File.class.getMethod("toPath");
+                Class<?> pathClass = Class.forName("java.nio.file.Path");
+                getFileStoreMethod = Class.forName("java.nio.file.Files").getMethod("getFileStore", pathClass);
+
+                mountEntryMethod = Class.forName("sun.nio.fs.UnixFileStore").getDeclaredMethod("entry");
+                hasOptionMethod =
+                        Class.forName("sun.nio.fs.UnixMountEntry").getDeclaredMethod("hasOption", String.class);
+
+                Throwable mountEntryThrowable = ReflectionUtil.trySetAccessible(mountEntryMethod);
+                Throwable hasOptionThrowable = ReflectionUtil.trySetAccessible(hasOptionMethod);
+                if (mountEntryThrowable != null || hasOptionThrowable != null) {
+                    if (!(mountEntryThrowable instanceof SecurityException) ||
```

Comment:
what exception are you expecting if a module accessibility is the root cause of the failure?